packages/transformers/js/core/src/dependency_collector.rs

```diff
@@ -458,10 +458,17 @@ impl<'a> Fold for DependencyCollector<'a> {
               // Promise.resolve().then(() => require('foo'))
               // Promise.resolve().then(() => { return require('foo') })
               // Promise.resolve().then(function () { return require('foo') })
+              //   but not
+              // Promise.resolve(require('foo'))
               if let Call(call) = &*member.obj {
                 if let Callee::Expr(e) = &call.callee {
                   if let Member(m) = &**e {
-                    if match_member_expr(m, vec!["Promise", "resolve"], self.decls) {
+                    if match_member_expr(m, vec!["Promise", "resolve"], self.decls) &&
+                      // Make sure the arglist is empty.
+                      // I.e. do not proceed with the below unless Promise.resolve has an empty arglist
+                      // because build_promise_chain() will not work in this case.                   
+                      !call.args.get(0).is_some()
```

Comment:
Is there a reason you didn't do it like this? I had to think for a moment when I first read `!(...).get(0).is_some()`
```suggestion
                      call.args.is_empty()
```

Reply:
I thought that it would catch ‘undefined’/‘null’ but that was probably stupid.

Will fix that when I get home from my vacation